exporter/exporterhelper/queued_retry.go

```diff
@@ -193,30 +134,35 @@ func (qrs *queuedRetrySender) onTemporaryFailure(logger *zap.Logger, req interna
 }
 
 // start is invoked during service startup.
-func (qrs *queuedRetrySender) start(ctx context.Context, host component.Host) error {
-	if err := qrs.initializePersistentQueue(ctx, host); err != nil {
-		return err
+func (qrs *queuedRetrySender) start(ctx context.Context, host component.Host, set exporter.CreateSettings) error {
+	if qrs.queue == nil {
+		return nil
 	}
 
-	qrs.queue.StartConsumers(qrs.queueSettings.config.NumConsumers, func(item internal.Request) {
-		_ = qrs.consumerSender.send(item)
-		item.OnProcessingFinished()
+	err := qrs.queue.Start(ctx, host, internal.QueueSettings{
+		CreateSettings: set,
+		DataType:       qrs.signal,
+		Callback: func(item internal.Request) {
+			_ = qrs.consumerSender.send(item)
+			item.OnProcessingFinished()
+		},
```

Comment:
Same as previous comment, better to pass all of these in the ctor if possible or at least pass only the callback.

Reply:
The idea is to expose the constructors to the user, as you suggested in https://github.com/open-telemetry/opentelemetry-collector/pull/8248#discussion_r1302106280. I wanted to reuse the same structures in the public and private packages. If we want to move all these arguments that users should not be providing, we will need different public structs encapsulating the internal ones. Also, we would likely need to change the signature of the Option from `type Option func(*baseSettings)` to `type Option func(*baseExporter)`, which should not be considered as a breaking change I believe. Let me try this out

Reply:
This involves too many additional changes. I would like to merge this one and try it in a separate PR